openadapt/scripts/scrub.py

```diff
@@ -0,0 +1,104 @@
+"""Module for scrubbing a media file.
+
+Usage: python -m openadapt.scripts.scrub <media_file_path>
+
+"""
+
+import os
+import sys
+
+from loguru import logger
+from PIL import Image
+import cv2
+import numpy as np
+
+from openadapt import scrub
+
+
+def scrub_mp4(mp4_file: str) -> str:
+    """
+    Scrub a mp4 file.
+
+    Args:
+        mp4_file: Path to the mp4 file.
+
+    Returns:
+        Path to the scrubbed (redacted) mp4 file.
+    """
+
+    scrubbed_file = mp4_file[:-4] + "_scrubbed.mp4"
+
+    cap = cv2.VideoCapture(mp4_file)
+    frame_width = int(cap.get(cv2.CAP_PROP_FRAME_WIDTH))
+    frame_height = int(cap.get(cv2.CAP_PROP_FRAME_HEIGHT))
+    fps = int(cap.get(cv2.CAP_PROP_FPS))
+    fourcc = cv2.VideoWriter_fourcc("m", "p", "4", "v")
+    out = cv2.VideoWriter(
+        scrubbed_file, fourcc, fps, (frame_width, frame_height)
+    )
+
+    while cap.isOpened():
+        # frameId = cap.get(1)  # current frame number
+        ret, frame = cap.read()
+        if not ret:
+            break
+        # Convert frame to PIL.Image
+        image = Image.fromarray(cv2.cvtColor(frame, cv2.COLOR_BGR2RGB))
```

Comment:
> Reason: it takes approx 1-2sec per Image to redact (on average). 1 Frame is 1 Image. And we have tons of Frames in a single video.

Thanks @KrishPatel13 ! Can you please measure how long each line takes to run so that we can identify the specific line that is the bottleneck? I suspect it may be this line.

Once we've identified the bottleneck we can look into whether it's possible to improve the performance, e.g. using gstreamer instead of cv2. On the other hand, this may be fine for now.

Reply:
Thank you @abrichr,

I will do an analysis as you have pointed out! before that I will address all of the comments below :)

Reply:
Line 71: `image = Image.fromarray(cv2.cvtColor(frame, cv2.COLOR_BGR2RGB))`
Line 76: `redacted_image = scrub.scrub_image(image)`
Line 81: `redacted_frame = cv2.cvtColor(np.array(redacted_image), cv2.COLOR_RGB2BGR)`
Line 87:` out.write(redacted_frame)`
<img width="425" alt="image" src="https://github.com/MLDSAI/OpenAdapt/assets/65433817/c252a08e-cd1f-42e4-b66a-944ec7f3bd56">

For a 2 second mp4 file, it took 106.14 seconds in total.
```
Processing: 100% | █████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████ | 61/61 |  1.74s/frame | [01:46<00:00] | 
Time spent on Line 71: 0.2766454219818115
Time spent on Line 76: 104.70847868919373
Time spent on Line 81: 0.3421134948730469
Time spent on Line 87: 0.514906644821167
Scrubbed .mp4 file saved to the relative path: sample2_scrubbed.mp4
(openadapt-py3.10) PS P:\OpenAdapt AI - MLDS AI\cloned_repo\OpenAdapt> 
```
Unit of measurement: seconds.

Reply:
Looks like majority of time is spent in redaction. 

Even if we use `FFMPEG`, it would not make any significant impact on performance as we will have to use our scrub)image function for redaction at the end of the day.

Reply:
Makes sense, thank you @KrishPatel13 !